asab/web/tenant/service.py

```diff
@@ -134,6 +156,37 @@ def install(self, web_container):
 		web_container.WebApp.on_startup.append(self._set_up_tenant_web_wrapper)
 
 
+	def is_ready(self) -> bool:
+		"""
+		Check if all tenant providers are ready.
+
+		Returns:
+			bool: Are all tenant providers ready?
+		"""
+		for provider in self.Providers:
+			if not provider.is_ready():
+				return False
+		return True
+
+
+	def set_ready(self, provider: TenantProviderABC):
```

Comment:
This is weird ... it should be called "check_ready, it should not require a tenant and mainly, it should handle edge (true->false, false->true) filtering internally.